app/controllers/spree/admin/active_sale_events_controller.rb

```diff
@@ -26,11 +29,20 @@ def update
         if params[:active_sale_event][:taxon_ids].present?
           params[:active_sale_event][:taxon_ids] = params[:active_sale_event][:taxon_ids].split(',')
         end
+
         super
       end
 
       private
 
+        def create_promotion
+          CreatePromotion.call(active_sale_event: @object)
```

Comment:
let's move the logic to ActiveSaleEventCreator/Updater service and remove after_action.